### PR TITLE
Misc code reshuffle

### PR DESCRIPTION
**Pull request by evelikov**
Pure code motion - no functional changes intended.

- stop mixing functions and sequential (?) code - keep functions at the top, everything else at the bottom
- separate between user-selectable options and internal variables - the `setup` movement
- couple other nitpicks

### CODE REVIEW

File: arch-chroot.in [hunk 1]
```diff
@@ -4,9 +4,6 @@ shopt -s extglob
 
 m4_include(common)
 
-setup=chroot_setup
-unshare=0
```
Comment (YHNdnzj) [inline arch-chroot.in:8]: Hmm, I think we have a coding style here to do global variable initialization first

See also: [mkinitcpio](https://github.com/archlinux/mkinitcpio/blob/master/mkinitcpio)

Comment (YHNdnzj) [reply arch-chroot.in:8]: Also I think the current order makes sense. You'll see var declaration/initialization first, then the code that uses them.

Comment (evelikov) [reply arch-chroot.in:8]: Well kind of .... unlike mkinitcpio we have a clunky (m4) include, which makes the final code a complete mess.
Should probably move that instead.

**Comment by YHNdnzj** [comment]
> Well kind of .... unlike mkinitcpio we have a clunky (m4) include, which makes the final code a complete mess.
Should probably move that instead.

Actually we may try to get rid of `m4` completely, to switch to source `common` in runtime instead. 

mkinitcpio is doing this already.

**Comment by evelikov** [comment]
> Actually we may try to get rid of m4 completely, to switch to source common in runtime instead.

The `common` is a complete misnomer since half of it isn't common at all. Generally having it a separate sourcable makes sense for something like mkinitcpio and less so here - there is a separate PR coming to unwrap that.

But overall one should really look at each of these things separately - as a man once said "divide and conquer"

**Comment by Foxboron** [comment]
Thanks!

**Pull request merged**